File: learn_cpp/backup_20251117_090625/01_basics/examples/04_control_flow.cpp
#include <iostream>
#include <string>

int main() {
    std::cout << "=== C++控制流演示 ===" << std::endl;
    
    // if-else语句
    std::cout << "\n--- if-else语句 ---" << std::endl;
    int score = 85;
    std::cout << "学生成绩: " << score << std::endl;
    
    if (score >= 90) {
        std::cout << "等级: A (优秀)" << std::endl;
    } else if (score >= 80) {
        std::cout << "等级: B (良好)" << std::endl;
    } else if (score >= 70) {
        std::cout << "等级: C (中等)" << std::endl;
    } else if (score >= 60) {
        std::cout << "等级: D (及格)" << std::endl;
    } else {
        std::cout << "等级: F (不及格)" << std::endl;
    }
    
    // switch-case语句
    std::cout << "\n--- switch-case语句 ---" << std::endl;
    char grade = 'B';
    std::cout << "等级: " << grade << std::endl;
    
    switch (grade) {
        case 'A':
            std::cout << "优秀！继续保持！" << std::endl;
            break;
        case 'B':
            std::cout << "良好！再接再厉！" << std::endl;
            break;
        case 'C':
            std::cout << "中等，需要努力！" << std::endl;
            break;
        case 'D':
            std::cout << "及格了，但要加油！" << std::endl;
            break;
        case 'F':
            std::cout << "不及格，需要重修！" << std::endl;
            break;
        default:
            std::cout << "无效的等级！" << std::endl;
            break;
    }
    
    // for循环
    std::cout << "\n--- for循环 ---" << std::endl;
    std::cout << "1到10的数字: ";
    for (int i = 1; i <= 10; ++i) {
        std::cout << i;
        if (i < 10) std::cout << ", ";
    }
    std::cout << std::endl;
    
    // 计算1到100的和
    int sum = 0;
    for (int i = 1; i <= 100; ++i) {
        sum += i;
    }
    std::cout << "1到100的和: " << sum << std::endl;
    
    // while循环
    std::cout << "\n--- while循环 ---" << std::endl;
    int countdown = 5;
    std::cout << "倒计时: ";
    while (countdown > 0) {
        std::cout << countdown << " ";
        --countdown;
    }
    std::cout << "发射！" << std::endl;
    
    // do-while循环
    std::cout << "\n--- do-while循环 ---" << std::endl;
    int number;
    do {
        std::cout << "请输入一个正数 (输入0退出): ";
        std::cin >> number;
        if (number > 0) {
            std::cout << "你输入的正数是: " << number << std::endl;
        } else if (number < 0) {
            std::cout << "这是负数，请输入正数!" << std::endl;
        }
    } while (number != 0);
    std::cout << "程序结束!" << std::endl;
    
    // 嵌套循环 - 打印乘法表
    std::cout << "\n--- 嵌套循环：九九乘法表 ---" << std::endl;
    for (int i = 1; i <= 9; ++i) {
        for (int j = 1; j <= i; ++j) {
            std::cout << j << "*" << i << "=" << (i * j);
            if (j < i) std::cout << "  ";
        }
        std::cout << std::endl;
    }
    
    // break和continue
    std::cout << "\n--- break和continue ---" << std::endl;
    std::cout << "找到第一个能被7整除的两位数: ";
    for (int i = 10; i < 100; ++i) {
        if (i % 2 == 0) {
            continue;  // 跳过偶数
        }
        if (i % 7 == 0) {
            std::cout << i << std::endl;
            break;  // 找到后退出循环
        }
    }
    
    // 范围for循环 (C++11)
    std::cout << "\n--- 范围for循环 (C++11) ---" << std::endl;
    int numbers[] = {1, 2, 3, 4, 5};
    std::cout << "数组元素: ";
    for (int num : numbers) {
        std::cout << num << " ";
    }
    std::cout << std::endl;
    
    std::string word = "C++";
    std::cout << "字符串 \"" << word << "\" 的字符: ";
    for (char c : word) {
        std::cout << c << " ";
    }
    std::cout << std::endl;
    
    return 0;
}

File: learn_go/examples/04_control_flow.go
package main

import "fmt"

func main() {
	fmt.Println("=== 控制流 ===")

	// 1. if 语句（Go 不需要括号）
	// Java:   if (x > 0) { ... }
	// Go:     if x > 0 { ... }
	x := 10
	if x > 0 {
		fmt.Println("x is positive")
	} else if x < 0 {
		fmt.Println("x is negative")
	} else {
		fmt.Println("x is zero")
	}

	// 2. if 初始化语句（Go 特色）
	// 在 if 中声明变量，作用域仅限于 if-else 块
	if num := 9; num < 0 {
		fmt.Println(num, "is negative")
	} else if num < 10 {
		fmt.Println(num, "has 1 digit")
	} else {
		fmt.Println(num, "has multiple digits")
	}

	// 3. for 循环（Go 只有 for，没有 while）
	// Java:   for (int i = 0; i < 5; i++) { ... }
	// Go:     for i := 0; i < 5; i++ { ... }
	fmt.Println("\n传统 for 循环:")
	for i := 0; i < 5; i++ {
		fmt.Printf("%d ", i)
	}
	fmt.Println()

	// 4. for 作为 while（Java 的 while 循环）
	// Java:   while (j < 3) { ... }
	// Go:     for j < 3 { ... }
	fmt.Println("\nfor 作为 while:")
	j := 0
	for j < 3 {
		fmt.Printf("%d ", j)
		j++
	}
	fmt.Println()

	// 5. 无限循环（需要 break 退出）
	// Java:   while (true) { ... }
	// Go:     for { ... }
	fmt.Println("\n无限循环（break 退出）:")
	count := 0
	for {
		count++
		if count > 3 {
			break
		}
		fmt.Printf("%d ", count)
	}
	fmt.Println()

	// 6. range 遍历（类似 Java 的增强 for，Kotlin 的 for-in）
	// Java:   for (int n : numbers) { ... }
	// Kotlin: for (n in numbers) { ... }
	// Go:     for i, n := range numbers { ... }
	fmt.Println("\nrange 遍历:")
	numbers := []int{10, 20, 30, 40}
	for i, n := range numbers {
		fmt.Printf("index: %d, value: %d\n", i, n)
	}

	// 7. switch 语句（Go 不需要 break，自动 break）
	// Java:   需要 break 防止 fall through
	// Go:     自动 break，需要 fallthrough 才继续
	fmt.Println("\nswitch 语句:")
	day := 3
	switch day {
	case 1:
		fmt.Println("Monday")
	case 2:
		fmt.Println("Tuesday")
	case 3:
		fmt.Println("Wednesday")
	case 4:
		fmt.Println("Thursday")
	case 5:
		fmt.Println("Friday")
	default:
		fmt.Println("Weekend")
	}

	// 8. switch 无表达式（类似 if-else 链）
	// Go 特色：switch 可以不带表达式
	score := 85
	switch {
	case score >= 90:
		fmt.Println("Grade: A")
	case score >= 80:
		fmt.Println("Grade: B")
	case score >= 70:
		fmt.Println("Grade: C")
	default:
		fmt.Println("Grade: F")
	}
}
